deep_qa/layers/attention/matrix_attention.py

```diff
@@ -87,10 +87,19 @@ def call(self, inputs, mask=None):
         it to work, but ``K.dot()`` breaks.
         """
         matrix_1, matrix_2 = inputs
-        num_rows_1 = K.int_shape(matrix_1)[1]
-        num_rows_2 = K.int_shape(matrix_2)[1]
+        matrix_1_shape = K.int_shape(matrix_1)
+        matrix_2_shape = K.int_shape(matrix_2)
+        num_rows_1 = matrix_1_shape[1]
+        num_rows_2 = matrix_2_shape[1]
         tiled_matrix_1 = K.repeat_elements(K.expand_dims(matrix_1, dim=2), num_rows_2, axis=2)
         tiled_matrix_2 = K.repeat_elements(K.expand_dims(matrix_2, dim=1), num_rows_1, axis=1)
+
+        # We need to be able to access K.int_shape() in compute_similarity() below, but in theano,
+        # calling a backend function makes it so you can't use K.int_shape() anymore.  Setting
+        # tensor._keras_shape here fixes that.
+        # pylint: disable=protected-access
+        tiled_matrix_1._keras_shape = matrix_1_shape[:2] + (num_rows_2,) + matrix_1_shape[2:]
```

Comment:
It's a total hack that we shouldn't rely on too extensively.  The better solution would be to fix all theano backend functions to set the `_keras_shape` on the returned tensor.  But I'm not in much of a mood to submit PRs to keras right now, and we're probably still in the PR freeze, anyway.